src/amo/sagas/collections.js

```diff
@@ -213,25 +229,77 @@ export function* updateCollection({
       `/${lang}/${clientApp}/collections/${user}/${effectiveSlug}/`
     ));
 
+    const slugWasEdited = slug && slug !== collectionSlug;
     if (!slugWasEdited) {
       // Invalidate the stored collection object. This will force each
       // component to re-fetch the collection. This is only necessary
       // when the slug hasn't changed.
       yield put(deleteCollectionBySlug(effectiveSlug));
     }
   } catch (error) {
-    log.warn(`Failed to update collection: ${error}`);
+    log.warn(`Failed to ${action} collection: ${error}`);
     yield put(errorHandler.createErrorAction(error));
     yield put(finishFormOverlaySubmit(formOverlayId));
   }
 }
 
+export function* createCollection({
+  payload: {
+    errorHandlerId,
+    defaultLocale,
+    description,
+    formOverlayId,
+    name,
+    slug,
+    user,
+  },
+}: CreateCollectionAction,
+_modifyCollection: typeof modifyCollection = modifyCollection,
```

Comment:
I designed `createCollection` and `updateCollection` to use dependency injection for `modifyCollection`, so we could test `modifyCollection` independently and just check that `createCollection` and `updateCollection` call it appropriately, but I couldn't figure out how to test `modifyCollection` independently using `sagaTester`.